metalarchives_importer.user.js

```diff
@@ -0,0 +1,175 @@
+// ==UserScript==
+// @name		Import Metal Archives releases into MB
+// @namespace	https://github.com/murdos/musicbrainz-userscripts/
+// @version		2015.08.23.4
+// @description	Add a button on Metal Archives release pages allowing to open MusicBrainz release editor with pre-filled data for the selected release
+// @downloadURL	https://raw.github.com/murdos/musicbrainz-userscripts/master/metalarchives_importer.user.js
+// @update		https://raw.github.com/murdos/musicbrainz-userscripts/master/metalarchives_importer.user.js
+// @include		http://www.metal-archives.com/albums/*/*/*
+// @require		https://ajax.googleapis.com/ajax/libs/jquery/1.3.2/jquery.js
+// @require		lib/mbimport.js
+// @require    	lib/logger.js
+// ==/UserScript==
+$(document).ready(function() {
+    var release = retrieveReleaseInfo();
+    insertLink(release);
+    LOGGER.info("Parsed release: ", release);
+});
+
+function setreleasedate(release, datestring) {
+    var patt = new RegExp('^d{4}$');
+    if (patt.exec(datestring)) {
+        release.year = datestring;
+    } else if (datestring.indexOf(',') != -1) {
+        var commaindex = datestring.indexOf(',');
+        var d = new Date(datestring.substring(0, commaindex - 2) + datestring.substring(commaindex));
+        release.year = d.getFullYear();
+        release.month = d.getMonth() + 1;
+        release.day = d.getDate();
+    } else {
+        var d = new Date("2 " + datestring);
+        release.year = d.getFullYear();
+        release.month = d.getMonth() + 1;
+    }
+    return release;
+}
+
+function getGenericalData() {
+    var rdata = new Array();
+    var keydata = $('dl.float_left dt, dl.float_right dt').map(function() {
+        var s = $.trim($(this).text());
+        return s.substring(0, s.length - 1);
+    }).get();
+    var valuedata = $('dl.float_left dd,dl.float_right dd').map(function() {
+        return $.trim($(this).text());
+    }).get();
+    for (i = 0; i < keydata.length; i++) {
+        rdata[keydata[i]] = valuedata[i];
+    }
+    return rdata;
+}
+
+function retrieveReleaseInfo() {
+    var release = new Object();
+    release.artist_credit = new Array();
+    release.artist_credit.push({
+        artist_name: $('h2.band_name').text(),
+        credited_name: $('h2.band_name').text()
+    });
+    release.title = $('h1.album_name').text();
+
+    var rdata = getGenericalData();
+
+    release = setreleasedate(release, rdata["Release date"]);
+    //todo add case for multiple labels if such a case exist
+    release.labels = new Array();
+    release.labels.push({
+        name: (rdata["Label"] == "Independent" ? "[no label]" : rdata["Label"]),
+        catno: (rdata["Catalog ID"] == "N/A" ? "" : rdata["Catalog ID"])
+    });
+
+    release.type = ReleaseTypes[rdata["Type"]][0];
```

Comment:
I would first check if ReleaseTypes has rdata["Type"], because if MA adds new Type it could break the script and better let the field empty and let the user import (and eventually report the missing type).